website/source/docs/auth/approle.html.md

```diff
@@ -0,0 +1,767 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AppRole"
+sidebar_current: "docs-auth-approle"
+description: |-
+  The AppRole backend allows machines and services to authenticate with Vault.
+---
+
+# Auth Backend: AppRole
+
+This backend allows machines and services (logically referred as `app`s) to
+authenticate with Vault, by registering them as AppRoles. The open design of
+AppRoles, enables a varied set of Apps to authenticate themselves. Since an
+AppRole can represent a service, or a machine or anything that can be IDed,
+this backend is a potential successor for the App-ID backend.
+
+### AppRole
+
+An AppRole represents a set of Vault policies, under a name. In essense, if a
+machine needs to authenticate with Vault for a set of policies, an AppRole can
+be registered under the machine's name with the desired set of policies. If a
+service requires a set of Vault policies, an AppRole can be registered under
+the service's name with the desired policies.  The credentials presented at the
+login endpoint depends on the constraints set on AppRoles.
+
+### RoleID 
+
+RoleID is a credential to be used at the login endpoint.  The credentials used
+to fetch a Vault token depends on the configured contraints on the AppRole.
+The credential `role_id` is a required argument for the login endpoint at all
+times. RoleIDs by default are unique UUIDs that map to the human read-able
+AppRole names. This credential lets the backend know which AppRole to refer to,
+in verifying the set constraints. RoleID for an AppRole can be fetched via
+`role/<role_name>/role-id` endpoint.
+
+### SecretID
+
+SecretID is a credential to be used at the login endpoint.  By default, this
+backend enables a login constraint on the AppRole, called `bind_secret_id`.
+When this constraint is enabled, the login endpoint expects another credential,
+`secret_id` to be presented, along with `role_id`. The backend supports both
+creation of SecretID by the backend and setting custom SecretID by the client.
+It is recommended that SecretIDs be generated by the backend. The ones
+generated by the backend will be cryptographically strong random UUIDs.
+SecretIDs have properties like usage-limit, TTLs and expirations; similar to
+tokens. SecretID for an AppRole can be fetched via `role/<role_name>/secret-id`
+endpoint.
+
+### Pull And Push SecretID Modes
+
+If the SecretID generated by the backend is fetched and used for login, it is
+referred as `Pull` mode. If a "custom" SecretID is set against an AppRole by
+the client, it is referred as a `Push` mode.
+
+While the `user_id` of the App-ID backend worked in a `Push` mode, this backend
+recommends the `Pull` mode. The `Pull` mode is supported in AppRole backend,
```

Comment:
@vishalnayak shouldn't this say "The `Push` mode is supported in AppRole backend *only* to be able to…"?

Or have I totally misunderstood things? :)